contract/services.py

```diff
@@ -1,41 +1,157 @@
 import core
+import json
 
+from django.db.models.query import QuerySet
+from django.contrib.auth.models import AnonymousUser
+from django.core.serializers.json import DjangoJSONEncoder
+from django.forms.models import model_to_dict
 
-@core.comparable
-class PolicyHolderContract(object):
+from contract.models import Contract as ContractModel, \
+    ContractDetails as ContractDetailsModel, \
+    ContractContributionPlanDetails as ContractContributionPlanDetailsModel
 
-    def __init__(self, policy_holder):
-        self.policy_holder = policy_holder
-        pass
+from policyholder.models import PolicyHolderInsuree
+from contribution_plan.models import ContributionPlanBundleDetails
 
-    def __eq__(self, other):
-        return isinstance(other, self.__class__) and self.__dict__ == other.__dict__
+
+def check_authentication(function):
+    def wrapper(self, *args, **kwargs):
+        if type(self.user) is AnonymousUser or not self.user.id:
+            return {
+                "success": False,
+                "message": "Authentication required",
+                "detail": "PermissionDenied",
+            }
+        else:
+            result = function(self, *args, **kwargs)
+            return result
+    return wrapper
+
+
+class Contract(object):
+
+    def __init__(self, user):
+        self.user = user
+
+    @check_authentication
+    def create(self, contract):
+        try:
+            c = ContractModel(**contract)
+            c.state = ContractModel.STATE_DRAFT
+            c.json_ext = json.dumps(_save_json_external(
+                user_id=self.user.id,
```

Comment:
Don't think it is required for create because the HistoryModel have date and user created

Reply:
Yes, you are right. I can try to use as you wrote - the data from HistoryModel.